src/main/java/com/esri/samples/graphics_update/position_sumulator/MessageGenerator.java

```diff
@@ -0,0 +1,191 @@
+        // singleton; only one listener
+        if (updateMessageEventRunner == null) {
+            updateMessageEventRunner = new UpdateMessageEventRunner(listener);
+        }
+    }
+
+    /**
+     * Method to start generating update messages for each vehicle
+     */
+    public void startMessages() {
+        TimerTask timerTask = new TimerTask() {
+            @Override
+            public void run() {
+                if (updateMessageEventRunner != null) {
+                    moveAllGraphics();
+                }
+            }
+        };
+
+        // timer to generate new messages on a separate thread every 20ms
+        timer = new Timer();
+        timer.schedule(timerTask,1000,20);
+    }
+
+    /**
+     * Method to iterate through all vehicles and provide an update message with a new position
+     */
+    private void moveAllGraphics() {
+        // loop through all the vehicles and increment the position along the route
+        int newPosition;
+        ArrayList<Point> routePoints;
+
+        for (Vehicle vehicle : vehicles) {
+
+            // new position along route
+            newPosition = vehicle.getPositionAlongRoute() + 1;
+
+            // get points for route
+            routePoints = routes.get(vehicle.getRouteID());
+
+            // check we've not reached end end of the route
+            if (newPosition < routePoints.size()) {
+                // update position to next point along route
+                vehicle.setPositionAlongRoute(newPosition);
+            } else {
+                // return to start of route
+                newPosition = 1;
+                vehicle.setPositionAlongRoute(newPosition);
+            }
+
+            // construct an update message
+            UpdateMessage updateMessage =
+                    new UpdateMessage(vehicle.getVehicleID(), routePoints.get(newPosition), vehicle.getStatus());
+
+            // send the message to the subscriber
+            UpdateMessageEvent event = new UpdateMessageEvent(this, updateMessage);
+            updateMessageEventRunner.run(event);
+        }
+    }
+
+    /**
+     * Method to stop vehicle update messages.  This method should be called when closing the JavaFX application
+     */
+    public void stopMessages() {
+        timer.cancel();
+    }
+
+    /**
+     * Method to read route information from CVS files contained in a data directory
+     */
+    private void ReadRouteFiles() {
+        int routeID = 1;
+
+        // loop through all the route files
+        File folder = new File("./data/");
+        File[] listOfFiles = folder.listFiles();
+
+        for (File file : listOfFiles) {
+            ArrayList<Point> path = new ArrayList();
+            String line = null;
+            String csvItem;
+            double xPos;
+            double yPos;
+
+            // open file
+            try {
+                BufferedReader reader = new BufferedReader(new FileReader(file));
```

Comment:
@Rachael-E and I learned this recently in a sample review - you could use try-with-resources to handle the creation and closing of the `BufferedReader`: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html